### DISCUSSION TITLE
Include a friendlier license like Apache 2.0?

### DISCUSSION THREAD

**Discussion by nandita-v** [indygreg/apple-platform-rs#112] [General]
Hi! For ease of using these crates in other projects, would it be possible to include a friendlier license such as Apache 2.0/MIT? Since the MPL 2.0 license is copy left, it isn't actually possible to seamlessly pull these crates in as dependencies in cargo.

**Comment by indygreg** [comment]
I'm willing to entertain the idea of changing the license from strictly MPL 2.0. However, it needs to be for good reasons.

I'm not convinced the reasons you stated are valid.

You are correct that MPL 2.0 is a copyleft license. However, it is *weakly* so.

MPL 2.0 components can be integrated into larger works. Unlike say the GPL, MPL 2.0's copyleft doesn't spread to other components. Rather, the copyleft applies to the MPL 2.0 components themselves. If you leave the MPL 2.0 source code unmodified, you include the license text and info on obtaining sources and you are pretty much done with it. If you modify the MPL 2.0 source code, then copyleft kicks in and you are required to publish those modifications.

I assert that an MPL 2.0 license is safe to pull in to larger Rust projects without fear of copyleft infringement on the larger work - assuming you don't modify source code.

Do you agree with my assessment? If so and you still feel that MPL 2.0 isn't appropriate, my next question is what modifications do you need to make to the MPL 2.0 licensed code and why can't you publish those modifications?
